core/controllers/profile_test.py

```diff
@@ -224,6 +224,7 @@ class ProfileLinkTests(test_utils.GenericTestBase):
     USERNAME = 'abc123'
```

Comment:
In all these tests I think you should do a self.swap() of urlfetch.fetch with a method that returns a deterministic string, so that running tests locally doesn't cause outgoing HTTP calls to be generated. The results of a test run should not depend on whether the machine has an internet connection.